server/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -402,6 +404,7 @@ private static ShardsAllocator createShardsAllocator(
             });
         }
         String allocatorName = SHARDS_ALLOCATOR_TYPE_SETTING.get(settings);
+        assert Version.CURRENT.major == Version.V_7_17_0.major + 1; // in v9 there is only one allocator
```

Comment:
I think we need to keep this property for as long we have a plugin API to supply a custom allocator implementation

https://github.com/elastic/elasticsearch/blob/8b92cd4bf59b2e7e99647308ad3d13cd247dbbf0/server/src/main/java/org/elasticsearch/cluster/ClusterModule.java#L385-L405

Reply:
Based on above I would suggest to emit the waring only when "balanced" is used.